std/process.d

```diff
@@ -3161,41 +3161,77 @@ static:
                 // and it is just as much of a security issue there.  Moreso,
                 // in fact, due to the case insensensitivity of variable names,
                 // which is not handled correctly by all programs.
-                if (name !in aa) aa[name] = toUTF8(envBlock[start .. i]);
+                auto val = toUTF8(envBlock[start .. i]);
+                if (name !in aa) aa[name] = val is null ? "" : val;
             }
         }
         else static assert(0);
         return aa;
     }
 
 private:
-    // Returns the length of an environment variable (in number of
-    // wchars, including the null terminator), or 0 if it doesn't exist.
-    version (Windows)
-    int varLength(LPCWSTR namez) @trusted nothrow
-    {
-        return GetEnvironmentVariableW(namez, null, 0);
-    }
-
     // Retrieves the environment variable, returns false on failure.
     bool getImpl(in char[] name, out string value) @trusted
     {
         version (Windows)
         {
+            // first we ask windows how long the environment variable is,
+            // then we try to read it in to a buffer of that length. Lots
+            // of error conditions because the windows API is nasty.
+
             import std.conv : to;
             const namezTmp = name.tempCStringW();
-            immutable len = varLength(namezTmp);
-            if (len == 0) return false;
+            WCHAR[] buf;
+
+            // clear error because GetEnvironmentVariable only says it sets it
+            // if the environment variable is missing, not on other errors.
+            SetLastError(NO_ERROR);
+            // len includes terminating null
+            immutable len = GetEnvironmentVariableW(namezTmp, null, 0);
+            if (len == 0)
+            {
+                immutable err = GetLastError();
+                if (err == ERROR_ENVVAR_NOT_FOUND)
+                    return false;
+                else // some other windows error. Might actually be NO_ERROR, because
+                     // GetEnvironmentVariable doesn't specify whether it sets on all
+                     // failures
+                    throw new Exception(sysErrorString(err));
```

Comment:
If this would prevent the function from being nothrow, better just return false.